lib/matplotlib/font_manager.py

```diff
@@ -273,11 +273,24 @@ def get_fontconfig_fonts(fontext='ttf'):
 
     fontfiles = {}
     try:
-        warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
         pipe = subprocess.Popen(['fc-list', '--format=%{file}\\n'],
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE)
-        output = pipe.communicate()[0]
+        # We emit a warning if the process has not completed after 5 seconds.
+        # We avoid showing it before since if this takes < 5 seconds, the user
+        # probably won't notice. However, this only works on Python 3 because
+        # the timeout keyword argument for communicate is not implemented in 2.
+        if six.PY2:
+            warnings.warn('Matplotlib is building the font cache using fc-list. '
+                          'This may take a moment.')
+            output = pipe.communicate()[0]
+        else:
+            try:
+                output = pipe.communicate(timeout=5)[0]
+            except subprocess.TimeoutExpired:
+                warnings.warn('Matplotlib is building the font cache using '
```

Comment:
Nitpick: I would put this message in a local variable so that it doesn't need to be spelled out twice in the code.